examples/lock-app/mbed/README.md

```diff
@@ -0,0 +1,175 @@
+<p align="center">
+  <img src="https://raw.githubusercontent.com/ARMmbed/mbed-os/master/logo.png" alt="ARM Mbed-OS logo"/>
+</p>
+
+<a name="main_title"></a>
+# CHIP ARM-MbedOS Lock Example Application
+
+The ARM-MbedOS Lock Example demonstrates how to remotely control a door lock
+device with one basic bolt.
+It uses buttons to test changing the lock and device states and LEDs to show 
+the state of these changes.  
+You can use this example as a reference for creating your own application.
+
+The example behaves as a CHIP accessory, that is a device that can be paired 
+into an existing CHIP network and can be controlled by this network.
+
+<hr>
+
+-   [CHIP ARM-MbedOS Lock Example Application](#main_title)
+    -   [Overview](#overview)
+    -   [Bluetooth LE advertising](#ble_advertising)
+    -   [Bluetooth LE rendezvous](#ble_rendezvous)
+    -   [Wi-Fi provisioning](#wifi_provisioning)
+
+-   [Building and Flashing Example Application](#build_and_flash)
+    -   [Development environment setup](#environment_setup)
+    -   [Building](#building)
+    -   [Flashing](#flashing)
+    -   [Testing the example](#testing_cli)
+    -   [Testing using CHIPTool](#testing_chiptool)
+    -   [Supported devices](#supported_devices)
+        -   [Notes](#notes)
+
+<hr>
+
+<a name="overview"></a>
+## Overview
+
+This example is running on the mbedos platform, which is based on 
+[ARM Mbed-OS system](#https://os.mbed.com/mbed-os/).
+
+The CHIP device that runs the lock application is controlled by the CHIP
+controller device over the Wi-Fi network. Some actions required before establishing full
+communication are described below.
+
+<a name="ble_advertising"></a>
+## Bluetooth LE advertising
+
+In this example, to commission the device onto a CHIP network, it must be
+discoverable over Bluetooth LE. The Bluetooth LE advertising starts
+automatically after device boot-up.
```

Comment:
I added new section which describes a little bit about button and LEDs states used in this example.
Because each HW board may have its own limitation in that area, I decided to put board specific information into the new 'Notes' column in 'Supported device' table. There we can find some info about capacitive button support and 'lock state' signal routed to external port.